src/Razor/test/Microsoft.AspNetCore.Razor.Test.ComponentShim/Microsoft.AspNetCore.Components/EventCallbackFactory.cs

```diff
@@ -16,12 +18,26 @@ public sealed class EventCallbackFactory
 
         public EventCallback Create(object receiver, Func<object, Task> callback) => default;
 
+        public EventCallback<T> Create<T>(object receiver, string callback) => default;
+
+        public EventCallback<T> Create<T>(object receiver, EventCallback<T> callback) => default;
+
         public EventCallback<T> Create<T>(object receiver, Action callback) => default;
 
         public EventCallback<T> Create<T>(object receiver, Action<T> callback) => default;
 
         public EventCallback<T> Create<T>(object receiver, Func<Task> callback) => default;
 
         public EventCallback<T> Create<T>(object receiver, Func<T, Task> callback) => default;
+
+        public EventCallback<T> CreateInferred<T>(object receiver, Action<T> callback, T value)
+        {
+            return Create(receiver, callback);
+        }
+
+        public EventCallback<T> CreateInferred<T>(object receiver, Func<T, Task> callback, T value)
```

Comment:
We need these for binding to components. The problem is that we can't directly call the `Create<T>` methods because we can't reliably write `T`. We have to let the C# compiler sort it out with overloading and type inference.